src/main/java/io/druid/segment/loading/DataSegmentPuller.java

```diff
@@ -10,10 +13,49 @@
 {
   /**
    * Pull down segment files for the given DataSegment and put them in the given directory.
+   * Deprecated in favor of URI pulling when possible
    *
    * @param segment The segment to pull down files for
-   * @param dir The directory to store the files in
+   * @param dir     The directory to store the files in
+   *
    * @throws SegmentLoadingException if there are any errors
    */
+  @Deprecated
   public void getSegmentFiles(DataSegment segment, File dir) throws SegmentLoadingException;
+
+  /**
+   * Pull down segment files for the given DataSegment and put them in the given directory.
+   *
+   * @param segmentUri The URI for the segment
+   * @param dir        The directory to store the files in
+   *
+   * @throws SegmentLoadingException if there are any errors
+   */
+  public void getSegmentFiles(URI segmentUri, File dir) throws SegmentLoadingException;
```

Comment:
I am trying to understand the need for this explicit method. It appears in the Query time lookup PR in druid that you added the ability to write a URI to a segment's loadSpec, and there are several methods that specifically check if an URI flag exists. There are also methods that only understand segment load specs. I think it may be worthwhile to make the loadSpec into an object, and use polymorphic deserialization to have URI and other types of load specs. We can then have each loadSpec implement their own way of getting segment files. Also, can you explain why we can't just pass around the segment object (versus using the loadSpec)